common/persistence/data_store_interfaces.go

```diff
@@ -969,6 +970,15 @@ func (d *DataBlob) GetEncodingString() string {
 	return string(d.Encoding)
 }
 
+func (d *DataBlob) GetVisibilityStoreInfo() ([]byte, common.EncodingType, error) {
+	data, err := json.Marshal(d)
+	if err != nil {
+		return nil, common.EncodingTypeEmpty, err
+	}
+
+	return data, common.EncodingTypeJSON, nil // for visibility, we use Json encode to keep it safe
```

Comment:
Are we always using encoding type json? From this method seems so. If it is true, we don't need to save it in search attribute

Reply:
Yes. David told me to add this because we are wrapping memo was another layer. And this encoding is for future reference for other people.